fix(canvas): `getRootElement` returns null during import

marstamm opened the pull request:
This PR fixes the following scenario:

1. In bpmn-js, we import all diagrams without setting an active root element
2. Once all imports are done, we set the root element (either a process or subprocess)
3. If an extensions reacts to `shape.add` during import and uses `canvas.getRootElement()`, an implicit root is created

Therefore, we do not want to create implicit roots when root elements are already present but not active.
We still ensure that we support simple use-cases, where no explicit root element is needed.
<!--

Thanks for creating this pull request!

Please make sure to link the issue you are closing as "Closes #issueNr". 
This helps us to understand the context of this PR.

-->

nikku commented on lib/core/Canvas.js:
```diff
@@ -515,14 +515,19 @@ Canvas.prototype.toggleMarker = function(element, marker) {
 };
 
 /**
- * Returns the current root element.
```
Could we rename the commit to more clearly indicate what it does?

> fix(canvas): do not implicitly create root if roots already exist

Could also add a description on how we have two distinct modes (nothing inbetween) 

* magic, implicit where an editor does not care about roots
* controlled where the toolkit building on top does care about roots and manages them explicitly (no magic at all)

nikku commented on test/spec/core/CanvasSpec.js:
```diff
@@ -545,6 +545,21 @@ describe('Canvas', function() {
+        var rootElement = canvas.getRootElement();
+
+        // then
+        expect(rootElement).to.not.exist;
```
```suggestion
        expect(rootElement).not.to.exist;
```

marstamm replied:
done with a5a6f49

marstamm commented:
I have to make a few adjustments in this PR:

- `getActiveLayer` expects that `getRootElement` always returns an element. It is now possible that we don't have an active layer

nikku commented:
How do you resolve this? :grey_question: :thinking:

marstamm commented:
My assumption: `getActiveLayer` can also return `null` in case no root element is active. This matches our decision for the root elements. I see the same argumentation we had for root elements: 

- either you build on top of a library that manages Roots: `null` is expected
- or you build with magic: an implicit root will be created

Alternative:
Return the base layer when no root is active. No major downsides for this one, as `activeLayer` is mostly used for previews/hover states, but would be inconsistent with how we handle active root elements.

nikku commented:
Assumption sounds reasonable to me, too.

Follow-up assumption: If components fail because of that they shall be architected in a safe way.

nikku commented:
But regarding your adjustments, please ensure we have the scenario / usage pattern that you're going to use covered in a spec.

marstamm commented:
It caught my eye because it breaks in a `viewbox()` call. I'll check if anything else breaks :+1:

marstamm commented:
The actual root cause why `viewbox()` broke was different then I thought: we did not clean up the active root reference when we removed the element (cf. 39f23f2d05c6a2496d0064a9dfd930ed47cc17e8). 

I fixed both the layer and root element cleanup. Ready for review again

nikku commented:
> The actual root cause why viewbox() broke was different then I thought: we did not clean up the active root reference when we removed the element :tada:

nikku reviewed (approved):
Let's regard this as a bug fix.

Do you mind updating the `BREAKING CHANGES` section for `v8.0.0` to mention the new behavior => Manual root elements vs automatic mode?

marstamm commented:
Will do
